sympy/core/symbol.py

```diff
@@ -737,3 +739,53 @@ def traverse(symbols, frame):
         del frame  # break cyclic dependencies as stated in inspect docs
 
     return syms
+
+def disambiguate(*iter):
+    """
+    Return a Tuple containing the passed expressions with distinct
+    symbols(that appear the same when printed) replaced with numerically
+    subscripted symbols.
+
+    Parameters
+    ==========
+
+    iter: list of symbols or expressions.
+
+    Examples
+    ========
+
+    >>> from sympy.core.symbol import disambiguate
+    >>> from sympy import Dummy, Symbol
+    >>> from sympy.abc import y
+
+    >>> tup = Symbol('_x'), Dummy('x'), Dummy('x')
+    >>> disambiguate(*tup)
+    (_x_2, _x, _x_1)
+
+    >>> disambiguate(Dummy('x'), Dummy('x'))
+    (_x, _x_1)
+
+    >>> t = Symbol('_x'), Dummy('x')/y
+    >>> disambiguate(*t)
+    (_x_1, _x/y)
+
+    >>> x = Symbol('x', integer=True)
+    >>> y = Symbol('x')
+    >>> disambiguate(x + y)
+    (x + x_1,)
+
+    """
+    new_iter = Tuple(*iter)
+    key = lambda x:tuple(sorted(x.assumptions0.items()))
+    syms = ordered(new_iter.free_symbols, keys=key)
+    mapping = {}
+    for s in syms:
+        mapping.setdefault(str(s), []).append(s)
+    reps = {}
+    for k in mapping:
+        reps[mapping[k][0]] = Symbol("%s"%(k), **mapping[k][0].assumptions0)
+        if len(mapping[k]) == 1:
+            continue
```

Comment:
I would move that line 786 to here before the "continue" then run the loop below "for i, s in enumnerate(mapping[k]): reps[s] = Symbol(...**s.assumptions0)` (prettier, however :-))

Reply:
Agreed, but the reason I placed there was it would output something like
`(_x_1, _x_0)`, I don't think we should have `0` subscripted elements(doesn't looks nice 😄). What do you say?